### PR TITLE
vim-patch:partial: 48c3f4e0bff7

### PR DESCRIPTION
**Pull request by clason**
Update runtime files
https://github.com/vim/vim/commit/48c3f4e0bff7efd289a7001b68c777b6f89a7057

partially skip `options.txt` (needs 9.0.0138)
omit terminal autochdir snippets (for now)

### CODE REVIEW

File: runtime/lua/vim/filetype/detect.lua [hunk 1]
```diff
@@ -940,7 +940,18 @@ function M.proto(bufnr, default)
   if getlines(bufnr, 2):find('.;$') then
     return 'cpp'
   else
-    return default
+    -- recognize Prolog by specific text in the first non-empty line
+    -- require a blank after the '%' because Perl uses "%list" and "%translate"
+    local line = nextnonblank(bufnr, 1)
+    if
+      line:find('<prolog>')
```
Comment (gpanders) [inline runtime/lua/vim/filetype/detect.lua:947]: `\<` and `\>` are word boundaries in Vim regex. Lua doesn't have an equivalent, so we have to fallback to regex here.

```suggestion
      matchregex(line, [[\<prolog\>]])
```

Comment (clason) [reply runtime/lua/vim/filetype/detect.lua]: Forgot that upstream just copy-pasted this from the Perl detection -- which is already ported, so I could do the same \o/

(Comment should have tipped me off -- I left it as is, just like upstream.)

File: runtime/doc/options.txt [hunk 2]
```diff
@@ -5780,6 +5781,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 	Name of the word list file where words are added for the |zg| and |zw|
 	commands.  It must end in ".{encoding}.add".  You need to include the
 	path, otherwise the file is placed in the current directory.
+	The path may include characters from 'isfname', space, comma and '@'.
```
Comment (seandewar) [inline runtime/doc/options.txt:5784]: I think this needs https://github.com/vim/vim/commit/bc49c5f48f89c2d6f4d88ee77f44a11d68293be3

File: runtime/autoload/dist/ft.vim [hunk 3]
```diff
@@ -358,7 +358,14 @@ func dist#ft#ProtoCheck(default)
...
     setf cpp
   else
-    exe 'setf ' . a:default
+    " recognize Prolog by specific text in the first non-empty line
+    " require a blank after the '%' because Perl uses "%list" and "%translate"
+    let l = getline(nextnonblank(1))
+    if l =~ '\<prolog\>' || l =~ '^\s*\(%\+\(\s\|$\)\|/\*\)' || l =~ ':-'
+      setf prolog
+    else
+      exe 'setf ' .. default
```
[the remark this answers is not in the document]
Comment (clason) [reply runtime/autoload/dist/ft.vim:367]: curse you, vim9script! you're at least six scripts ahead of me...

**Review by seandewar — APPROVED** [review]

**Review by zeertzjq — APPROVED** [review]
partial?

**Comment by clason** [comment]
> partial?

I assumed the terminal autochdir snippets were N/A (and apparently have bugs)?

**Comment by zeertzjq** [comment]
Oh wait. This change is applicable:
```
When a script has been processed to the end, the local variables declared
there will not be deleted.  Functions defined in the script can use them.
Example:
```

**Comment by zeertzjq** [comment]
And since this is missing the `'isfname'` doc, this is still partial.

**Comment by zeertzjq** [comment]
The partial skip is in `options.txt`.

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**